### PR TITLE
Add release note and r1.json entry for release 2201.4.2

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose
$subject
> Fixes #

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - /downloads/swan-lake-release-notes/2201.4-2/
+    - /downloads/swan-lake-release-notes/2201-4-2-swan-lake/
+    - /downloads/swan-lake-release-notes/2201-4-2-swan-lake
+---
+
+## Overview of Ballerina Swan Lake 2201.4.2
+
+<em>Swan Lake 2201.4.2 is the second patch release of Ballerina 2201.4.0 (Swan Lake Update 4) and it includes a new set of improvements to the language and developer tooling.</em>
+
+## Update Ballerina
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:17]:
```suggestion
## Update Ballerina
```
We need to verify this section @udda1996 @keizer619

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+To view bug fixes, see the [GitHub milestone for 2201.4.2 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.4.2+is%3Aclosed+label%3AType%2FBug).
+
+## Developer tools updates
+
+### Bug fixes
+
+To view bug fixes, see the GitHub milestone for 2201.4.2 (Swan Lake) of the repositories below.
+
+- [OpenAPI](https://github.com/ballerina-platform/openapi-tools/issues?q=is%3Aissue+milestone%3A%22Swan+Lake+2201.4.2%22+is%3Aclosed+)
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:47]:
```suggestion
- [OpenAPI](https://github.com/ballerina-platform/openapi-tools/issues?q=is%3Aissue+milestone%3A%22Swan+Lake+2201.4.2%22+is%3Aclosed+)
```
We can add in the general format if we have only one link.

Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:47]:
```suggestion
- [OpenAPI](https://github.com/ballerina-platform/openapi-tools/issues?q=is%3Aissue+milestone%3A%22Swan+Lake+2201.4.2%22+is%3Aclosed+)
```
the link is incorrect. Need to filter the bugs.

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Standard library updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.4.2 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/milestone/28?closed=1).
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:53]:
```suggestion
To view bug fixes, see the [GitHub milestone for 2201.4.2 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/milestone/28?closed=1).
```
Can we be consistent with the others and get a milestone link like below for standard library also? We have done this correctly in the [2201.3.3 release note](https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.3.3#standard-library-updates).

`https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aissue+milestone%3A2201.3.3+label%3AType%2FBug+is%3Aclosed+`

**Review by praneesha — CHANGES_REQUESTED** [review]

File: utils/rl.json [hunk 4]
```diff
@@ -22,6 +22,14 @@
                     "url": "/downloads/swan-lake-release-notes/swan-lake-2201.5.0",
                     "id": "swan-lake-2201.5.0"
                 },
+                {
+                    "dirName": "2201.4.2 (Swan Lake Update 4)",
```
Comment (praneesha) [inline utils/rl.json:26]:
```suggestion
                    "dirName": "2201.4.2 (Swan Lake Update 4)",
```
We need to add 2201.4.1 to the archived versions page also @sahanHe.

**Review by praneesha — CHANGES_REQUESTED** [review]

File: utils/rl.json (hunk 4 above)
Comment (sahanHe) [reply utils/rl.json:26]: it's already there right? https://github.com/ballerina-platform/ballerina-dev-website/blob/ee658383eaeaed006e91640a4143de448a5a35ab/utils/archived-lm.json#L26

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md (hunk 1 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:17]: Please change as per https://github.com/ballerina-platform/ballerina-dev-website/pull/6784.

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md (hunk 3 above)
Comment (sahanHe) [reply downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md]: updated in [7aa8117](https://github.com/ballerina-platform/ballerina-dev-website/pull/6780/commits/7aa81171ac6200b2d7d80eb3ce51242f28c64d66)

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md (hunk 2 above)
Comment (sahanHe) [reply downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md]: updated in [7aa8117](https://github.com/ballerina-platform/ballerina-dev-website/pull/6780/commits/7aa81171ac6200b2d7d80eb3ce51242f28c64d66)

Comment (sahanHe) [reply downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md]: updated in [7aa8117](https://github.com/ballerina-platform/ballerina-dev-website/pull/6780/commits/7aa81171ac6200b2d7d80eb3ce51242f28c64d66)

File: downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md (hunk 1 above)
Comment (sahanHe) [reply downloads/swan-lake-release-notes/swan-lake-2201.4.2/RELEASE_NOTE.md:17]: updated in [7aa8117](https://github.com/ballerina-platform/ballerina-dev-website/pull/6780/commits/7aa81171ac6200b2d7d80eb3ce51242f28c64d66)

**Pull request merged**